src/utils/crypto_utils/src/crypto_lib.c

```diff
@@ -242,16 +243,19 @@ CryptoKeyHandle RSAKey_ObjFromModulusBytesExponentInt(const uint8_t* N, size_t N
         goto done;
     }
 
-    result = pkey;
+    success = true;
+
 done:
 
-    if (result == NULL)
+    if (!success)
     {
         BN_free(rsa_N);
         BN_free(rsa_e);
+        EVP_PKEY_free(pkey);
```

Comment:
line 203 has RSA_new().  Seems like need a corresponding RSA_free() may be needed based on the docs: https://www.openssl.org/docs/manmaster/man3/RSA_new.html